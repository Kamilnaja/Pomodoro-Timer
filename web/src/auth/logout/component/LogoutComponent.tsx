import './logoutComponent.scss';
import { LogoutComponentProps } from './LogoutComponentProps';

export const LogoutComponent = (props: LogoutComponentProps) => (
  <div className="logout">
    <p className="logout__question">Are you sure to logout?</p>
    <div className="logout__buttons-wrapper">
      <button className="logout__button logout__button--no" onClick={() => props.handleCancel()}>
        No
      </button>
      <button className="logout__button logout__button-yes" onClick={() => props.handleLogout()}>
        Yes
      </button>
    </div>
  </div>
);
