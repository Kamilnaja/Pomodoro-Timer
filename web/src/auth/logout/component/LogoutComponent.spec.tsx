import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { handleCancel, handleLogout } from '../../testing/auth.testing.data';
import { LogoutComponent } from './LogoutComponent';

let logoutComponent: any;

describe('LogoutComponent', () => {
  beforeEach(() => {
    logoutComponent = shallow(<LogoutComponent handleLogout={handleLogout} handleCancel={handleCancel} />);
  });

  it('Should match snapshot', () => {
    expect(shallowToJson(logoutComponent)).toMatchSnapshot();
  });

  it('Should have logout wrapper', () => {
    expect(logoutComponent.find('.logout')).toHaveLength(1);
    expect(logoutComponent.find('.logout__question')).toHaveLength(1);
  });

  it('Should start handleCancel method after clicking in handleCancel button', () => {
    logoutComponent.find('.logout__button--no').simulate('click');
  });
});
