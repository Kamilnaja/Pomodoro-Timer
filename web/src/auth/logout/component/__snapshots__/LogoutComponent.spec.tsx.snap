// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutComponent Should match snapshot 1`] = `
<div
  className="logout"
>
  <p
    className="logout__question"
  >
    Are you sure to logout?
  </p>
  <div
    className="logout__buttons-wrapper"
  >
    <button
      className="logout__button logout__button--no"
      onClick={[Function]}
    >
      No
    </button>
    <button
      className="logout__button logout__button-yes"
      onClick={[Function]}
    >
      Yes
    </button>
  </div>
</div>
`;
